src/featureManager.ts

```diff
@@ -75,14 +73,15 @@ export class FeatureManager {
         return !shortCircuitEvaluationResult;
     }
 
+    async #getFeatureFlag(featureName: string): Promise<any> {
+        const featureFlag = await this.#provider.getFeatureFlag(featureName);
+        validateFeatureFlag(featureFlag);

```

Comment:
Let's move this to feaetureProvider. Decouple the feature flag validation from feature manager. (replied [here](https://github.com/microsoft/FeatureManagement-JavaScript/pull/17#discussion_r1830196943))